src/axolotl/utils/config.py

```diff
@@ -527,6 +527,22 @@ def validate_config(cfg):
     if cfg.fsdp and "bnb" in cfg.optimizer:
         raise ValueError(f"FSDP not compatible with {cfg.optimizer}")
 
+    if cfg.datasets:
+        for idx, ds_cfg in enumerate(cfg.datasets):
+            if (
+                ds_cfg.type
+                in [
+                    "ultra_apply_chatml",
+                    "toxic_apply_chatml",
```

Comment:
I don't think we ever had `toxic_apply_chatml` in our previous dpo release pre-cleanup.

Reply:
At least from how I've understood the DPO (and this took me some time, and I might have misunderstood it):
```yaml
datasets:
    path: argilla/ultrafeedback-binarized-preferences-cleaned
    split: train
    type: ultra_apply_chatml
   conversation: chatml 
```
This doesn't make any sense. You shouldn't set conversation here since the type actually also includes formatting for a ChatML conversation. In some cases, type serves both as the input format and the format used for conversation style, to my understanding. Therefore, I thought it would be valid to raise an error if you use a type format that also includes a conversation format and your setting conversation. 

Regarding toxic_apply_chatml  do you mean it is not supported or was previously not supported?

Reply:
@winglian, I now understand your point. However, I still believe that setting conversations for DPO should not be possible since it's done in type. Or am I still overlooking something?